### ISSUE TITLE
How do I build this tool?

### ISSUE THREAD

**Issue by zdavatz** [sapstern/-edifactconverter-#1]
Can you please explain how I build this tool?

**Comment by sapstern** [comment]
Did you read the wiki?

https://github.com/sapstern/-edifactconverter-/wiki

**Comment by zdavatz** [comment]
No, I was only looking in the README. Thanks! 

Do you support OpenTrans 2.1 as well?

**Comment by sapstern** [comment]
No, this is not supported.

But you can do it like this:

1) Use the converter to transform to EDI-XML
2) Use XSLT or Java Sax/DOM Parser to transform from EDI-XML to target XML

**Comment by zdavatz** [comment]
> Use XSLT or Java Sax/DOM Parser to transform from EDI-XML to target XML

Which Java Sax/DOM Parser would you recommend?

**Comment by sapstern** [comment]
I always use SAX:
import org.xml.sax.Attributes;
import org.xml.sax.SAXException;
import org.xml.sax.helpers.DefaultHandler;

Write a Class which extends DefaultHandler.

It fires events while traversing the EDI-XML tree.

Have a look to my code inside the repo:

com.sapstern.openedifact.sax.EdifactSaxParserToFlat

**Comment by zdavatz** [comment]
you mean here: https://github.com/sapstern/-edifactconverter-/blob/0357a7a5a9352b725777bfb34a3c4ed9b5e0aebb/openedifact-converter/src/main/java/com/sapstern/openedifact/sax/EdifactSaxParserToFlat.java

**Comment by zdavatz** [comment]
How much work do you think it would it take to add support of OpenTrans 2.1 to this tool?

**Comment by sapstern** [comment]
yes, I meant this class, have a look for the @Override annotations.

These are the methods which will be called by the sax parser.

To your last question. 

Not insignificant as it looks.

The main task is to apply the process know how to the mappings, I would think.

You need to know which EDIFACT field has its equivalent in the OpenTrans format.

**Comment by zdavatz** [comment]
> yes corret. 
https://github.com/sapstern/-edifactconverter-.git

**Comment by zdavatz** [comment]
Would it be easier if we would just start with a couple of sample messages?

**Comment by zdavatz** [comment]
Ok, build on Linux was successful! Now I am reading your next steps.

**Comment by sapstern** [comment]
yes, work with sample messages is a good starting point

**Comment by zdavatz** [comment]
Ok, I am now at step

1. Use the converter to transform to EDI-XML

How do I convert to EDI-XML?

At the moment I get:

```
~/.software/edifactconverter/openedifact-xsd-libs-96A/target> java -jar openedifact-xsd-libs-96A-1.1-SNAPSHOT.jar 
kein Hauptmanifestattribut, in openedifact-xsd-libs-96A-1.1-SNAPSHOT.jar
```

**Comment by sapstern** [comment]
Use openedifact-converter-2.2-SNAPSHOT.jar,  this holds the converter.

You can't  use java -jar, you just have to put the archives on the classpath and wirite your own class which calls the converter.
I think inside the test section of the maven project, there is an example.

**Comment by zdavatz** [comment]
This I do not understand from the Wiki:
```
com.openedi.sax.EdifactSaxParserToXML theXmlParser = EdifactSaxParserToXML .factory("UTF-8"); "<Encoding e.g. UTF-8" );
```
I am trying the following:

`
java -jar openedifact-converter/target/openedifact-converter-2.2-SNAPSHOT.jar openedifact-xsd-libs-96A/target/classes/EDIFACTINTERCHANGE_ORDERS_96A.xsd /tmp/SND_PLICA_ORDERS_134747_58408847
`

**Comment by zdavatz** [comment]
you mean here: `openedifact-sample-01` - I am looking at the sample now. Thank you!

**Comment by sapstern** [comment]
Example:
EdifactSaxParserToXML theXmlParser = EdifactSaxParserToXML.factory("UTF-8", logger);

theXmlParser.parse(Your Edifact as InputSource)

**Comment by sapstern** [comment]
Write a class with a main method and call the code

**Comment by zdavatz** [comment]
In which directory do I find the sample?

**Comment by sapstern** [comment]
Have a look to src/test/java/EDIFACT_2_XML_Test.java 

it instantiates and calls the parser.

**Comment by zdavatz** [comment]
this one: https://github.com/sapstern/-edifactconverter-/blob/master/openedifact-converter/src/test/java/com/sapstern/openedifact/segmenttest/EDIFACT_2_XML_Test.java

**Comment by zdavatz** [comment]
I prefer this one, it is simpler: https://github.com/sapstern/-edifactconverter-/blob/master/openedifact-sample-01/src/main/java/com/sapstern/openedifact/Sample01.java

**Comment by zdavatz** [comment]
How do I build this, so I can run it from commandline? 

`https://github.com/sapstern/-edifactconverter-/blob/master/openedifact-converter/src/test/java/com/sapstern/openedifact/segmenttest/EDIFACT_2_XML_Test.java`

**Comment by zdavatz** [comment]
I am now trying: `mvn package`

**Comment by sapstern** [comment]
You need to setup a java project, the whole stuff is meant to be used in calling the converter from your self written java code.

It is not running out of the box, sorry.

If you are not familiar with java, then it will be difficult, I am afraid.

**Comment by zdavatz** [comment]
I think the usage of the project would be greatly increased if it would come with a small commandline utility at actually uses the converter as you described above. See #2
